clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -12,23 +9,23 @@
     private ITransport _transport;
     private OnReceiveCallBack callback;
     private HashMap<String, Action> handlers = new HashMap<>();
-    private JsonParser jsonParser = new JsonParser();
-    private static final String RECORD_SEPARATOR = "\u001e";
+    private IHubProtocol protocol;
 
     public Boolean connected = false;
 
     public HubConnection(String url) {
         _url = url;
+        protocol = new JsonHubProtocol();
         callback = (payload) -> {
-            String[] messages = payload.split(RECORD_SEPARATOR);
 
-            for (String splitMessage : messages) {
+            InvocationMessage[] messages = protocol.parseMessages(payload);
 
-                // Empty handshake response "{}". We can ignore it
-                if (splitMessage.length() == 2) {
-                    continue;
+            // message will be null if we receive any message other than an invocation.
+            // Adding this to avoid getting error messages on pings for now.
+            for (InvocationMessage message: messages) {
+                if (message != null && handlers.containsKey(message.target)) {
+                    handlers.get(message.target).invoke(message.arguments[0]);
```

Comment:
how about methods taking more than one arguments?

Reply:
Haven't added support for that just yet 😄